internal/encryption/keyprovider/openbao/config.go

```diff
@@ -0,0 +1,104 @@
+package openbao
+
+import (
+	"errors"
+	"fmt"
+
+	openbao "github.com/openbao/openbao/api"
+	"github.com/opentofu/opentofu/internal/encryption/keyprovider"
+)
+
+type Config struct {
+	Address string `hcl:"address,optional"`
+	Token   string `hcl:"token,optional"`
```

Comment:
I kinda wonder how `token` should be managed? Ideally we'd like not to check it in to wherever the configuration lives, but the other fields could be. The [`api/auth` package](https://pkg.go.dev/github.com/openbao/openbao/api@main#section-directories) has various auth providers that perhaps could be detected and configured automatically to fetch & renew the token (e.g., AppRole, LDAP, ...) But I suppose Tofu would be a short-running process and so would be unlikely to need to renew the token in the first place?

Perhaps it should be read from the environment instead? Or something like [OpenBao Proxy](https://janma.github.io/openbao/docs/agent-and-proxy/proxy/apiproxy/) could be recommended instead, where Tofu would not maintain a token and instead contact a local auto-authing transparent proxy?

Some thoughts!

Reply:
Tbh, I think it you just make it optional, you can probably use `{BAO,VAULT}_{TOKEN,ADDR}` transparently.

Reply:
Ah, I see, you're doing that but overriding with the config. I see down below intbhe docs that you're marking token and address both as required, so I thought you weren't using the environment variables. :D

Reply:
Yes, token could be set via `BAO_TOKEN` environment variable and in such case we don't require it in config. However, it was not possible to omit it entirely so I made it completely optional here: [e710e8f](https://github.com/opentofu/opentofu/pull/1436/commits/e710e8f6646771eac7b328c1543c9caad759fb9b). It will allow OpenTofu to be used with OpenBao proxy configured.

Also, we could implement more auth options in the future and limit it to token / proxy from now on.

Thanks for the suggestion!